### DISCUSSION TITLE
setup() got an unexpected keyword argument 'fold_strategy'

### DISCUSSION THREAD

**Discussion by lgpatruyo** [pycaret/pycaret#1388] [Q&A] [answered]
Hi, I am trying to follow the code in your document "Time Series Forecasting with PyCaret Regression Module". 
I have installed PyCaret 2.2 and also the nightly version. I did create a separate virtual environment. 

I am getting the above error: "setup() got an unexpected keyword argument 'fold_strategy'". I actually do not see that argument in the method help. Any idea what could be wrong?

Setup() was not recognizing the argument 'test_data' either. I fixed that by only using (data = data, train_size = 0.9, data_split_shuffle = False,...) and deleting the 'test_data' argument.

Thanks,

**Comment by ngupta23** [comment, marked as answer]
Can you upgrade to the latest version of pycaret?

**Reply by lgpatruyo** [reply]
I think the problem was that I had two versions of Pycaret installed on my virtual environment: the 2.2 version and the nightly version. I uninstalled the 2.2 version (pip uninstall pycaret) and the problem was solved!
Thanks

**Comment by Marx-wrld** [comment]
This is a version issue. If you've installed the PyCaret 3.0 release candidate, you will find that the silent argument was removed. The documentation, stable release 2.3, has the silent argument.

Uninstalling your current PyCaret and installing 2.3 should fix your issue.

**Comment by gfragi** [comment]
Hi I have the same issue with rc5.

from pycaret.utils import version
version()
'3.0.0.rc5'

My setup is 
exp = setup(data=df['LeaseContractLength'], fh=12, fold=3, session_id=123)

and a head of my df is the following:

LeaseContractLength | PurchaseOption | vCPU | Memory | networkPerformance | StorageSize | operatingSystem_RHEL_HA | operatingSystem_Windows | DiskType_SSD | Month | Year | const
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
-0.105602 | -0.152916 | 12.833949 | -37.737905 | 0.045322 | 25.666031 | 9.778882 | 0.159119 | 9.778882 | 5.0 | 2018.0 | NaN
-0.108889 | -0.155410 | 12.891298 | -37.853456 | 0.042819 | 25.735064 | 9.800649 | 0.155486 | 9.800649 | 6.0 | 2018.0 | NaN
-0.112176 | -0.157904 | 12.948646 | -37.969008 | 0.040316 | 25.804097 | 9.822416 | 0.151854 | 9.822416 | 7.0 | 2018.0 | NaN
-0.115463 | -0.160397 | 13.005994 | -38.084560 | 0.037812 | 25.873130 | 9.844182 | 0.148222 | 9.844182 | 8.0 | 2018.0 | NaN


<br class="Apple-interchange-newline">

The index is continuous date (freq M)

![image](https://user-images.githubusercontent.com/9120111/222930106-e553ee25-610b-440f-a77b-4f84e2571e1b.png)

**Reply by gfragi** [reply]
I have update to rc9 and I dont have the issue.
